jwst/resample/resample.py

```diff
@@ -87,93 +90,317 @@ def __init__(self, input_models, output=None, single=False, blendheaders=True,
-                    img.data -= blevel
+                    data = img.data - blevel
+                else:
+                    data = img.data
 
-                inwht = resample_utils.build_driz_weight(img,
-                                                         weight_type=self.weight_type,
-                                                         good_bits=self.good_bits)
-                driz.add_image(img.data, img.meta.wcs, inwht=inwht,
-                               expin=img.meta.exposure.exposure_time)
+                driz.add_image(data, img.meta.wcs, inwht=inwht)
+
+            self.output_models.append(output_model)
 
-            # Update some basic exposure time values based on all the inputs
-            output_model.meta.exposure.exposure_time = texptime
-            output_model.meta.exposure.start_time = min(exposure_times['start'])
-            output_model.meta.exposure.end_time = max(exposure_times['end'])
-            output_model.meta.resample.product_exposure_time = texptime
-            output_model.meta.resample.weight_type = self.weight_type
-            output_model.meta.resample.pointings = pointings
+        return self.output_models
 
-            self.update_fits_wcs(output_model)
+    def resample_many_to_one(self):
+        """Resample and coadd many inputs to a single output.
 
-            self.output_models.append(output_model)
+        Used for stage 3 resampling
+        """
+        output_model = self.blank_output.copy()
+        output_model.meta.filename = self.output_filename
+        output_model.meta.resample.weight_type = self.weight_type
+        output_model.meta.resample.pointings = len(self.input_models.group_names)
+
+        if self.blendheaders:
+            self.blend_output_metadata(output_model)
+
+        # Initialize the output with the wcs
+        driz = gwcs_drizzle.GWCSDrizzle(output_model, pixfrac=self.pixfrac,
+                                        kernel=self.kernel, fillval=self.fillval)
+
+        for img in self.input_models:
+            inwht = resample_utils.build_driz_weight(img,
+                                                     weight_type=self.weight_type,
+                                                     good_bits=self.good_bits)
+            # apply sky subtraction
+            blevel = img.meta.background.level
+            if not img.meta.background.subtracted and blevel is not None:
+                data = img.data - blevel
+            else:
+                data = img.data
+
+            driz.add_image(data, img.meta.wcs, inwht=inwht)
+
+        # Resample variances array in self.input_models to output_model
+        self.resample_variance_array("var_rnoise", output_model)
+        self.resample_variance_array("var_poisson", output_model)
+        self.resample_variance_array("var_flat", output_model)
+        output_model.err = np.sqrt(output_model.var_rnoise + output_model.var_poisson
+                                   + output_model.var_flat)
```

Comment:
So you're not bothering to resample any of the variance arrays in the `many_to_many` version above, because that's only used for something like `outlier_detection` which doesn't need the resampled variance arrays for anything?

Reply:
Right, exactly.  This was a nice surprise of doing the refactor.  This speeds up outlier detection by a factor of ~3 from the alternative.